packages/blade/src/components/Input/OTPInput/__tests__/OTPInput.web.test.tsx

```diff
@@ -0,0 +1,143 @@
+    const onChange = jest.fn();
+    const user = userEvent.setup();
+    const otp = '123456';
+
+    const { getAllByLabelText } = renderWithTheme(
+      <OTPInput label={label} name="otp" onChange={onChange} />,
+    );
+
+    const allInputs = getAllByLabelText(/character/);
+    await user.type(allInputs[0], otp);
+
+    // should be called for each keystroke
+    expect(onChange).toHaveBeenCalledTimes(otp.length);
+    expect(onChange).toHaveBeenLastCalledWith({ name: 'otp', value: otp });
+  });
+
+  it('should handle onOTPFilled', async () => {
+    const label = 'Enter OTP';
+    const onOTPFilled = jest.fn();
+    const user = userEvent.setup();
+    const otp = '123456';
+
+    const { getAllByLabelText } = renderWithTheme(
+      <OTPInput label={label} name="otp" onOTPFilled={onOTPFilled} />,
+    );
+
+    const allInputs = getAllByLabelText(/character/);
+    await user.type(allInputs[0], otp);
+
+    expect(onOTPFilled).toHaveBeenLastCalledWith({ name: 'otp', value: otp });
+  });
+
+  it('should set value as an uncontrolled input', async () => {
+    const user = userEvent.setup();
+    const label = 'Enter OTP';
+    const otp = '123456';
+
+    const { getAllByLabelText } = renderWithTheme(<OTPInput label={label} name="otp" />);
+
+    const allInputs = getAllByLabelText(/character/);
+    await user.type(allInputs[0], otp);
+    allInputs.forEach((input, index) => {
+      expect(input).toHaveValue(Array.from(otp)[index]);
+    });
+  });
+
+  it('should set value as a controlled input', async () => {
+    const user = userEvent.setup();
+    const label = 'Enter OTP';
+    const otp = '123456';
+
+    const ControlledInputExample = (): ReactElement => {
+      const [value, setValue] = useState<string | undefined>('');
+
+      return <OTPInput label={label} value={value} onChange={({ value }) => setValue(value)} />;
+    };
+
+    const { getAllByLabelText } = renderWithTheme(<ControlledInputExample />);
+
+    const allInputs = getAllByLabelText(/character/);
+
+    await user.type(allInputs[0], otp);
+
+    allInputs.forEach((input, index) => {
+      expect(input).toHaveValue(Array.from(otp)[index]);
+    });
+  });
+
+  it('should pass a11y', async () => {
+    const { getAllByLabelText } = renderWithTheme(
+      <OTPInput
+        label="Enter OTP"
+        helpText="Enter OTP sent to your mobile"
+        validationState="none"
+        value="1234"
+        otpLength={4}
+      />,
+    );
+
+    const allInputs = getAllByLabelText(/character/);
+    allInputs.forEach((input) => {
+      expect(input).toBeRequired();
+      expect(input).toBeValid();
+      expect(input).toBeEnabled();
+    });
+    // asserting inside a loop leads to an issue with multiple parallel runs ref: https://github.com/dequelabs/axe-core/issues/3426
+    await assertAccessible(allInputs[0]);
+    await assertAccessible(allInputs[1]);
+    await assertAccessible(allInputs[2]);
+    await assertAccessible(allInputs[3]);
+  });
+});
```

Comment:
Optional: you can add another test for testing the focus management of individual inputs with arrow / tab keys

Reply:
Done! Thanks for the catch. Missed adding these 🙈 

Also added tests for backspace and delete key behaviours